src/ContinuousDynamicalSystem/include/BipedalLocomotion/ContinuousDynamicalSystem/FloatingBaseSystemKinematics.h

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file FloatingBaseSystemKinematics.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTINUOUS_DYNAMICAL_SYSTEM_FLOATING_BASE_SYSTEM_KINEMATICS_H
+#define BIPEDAL_LOCOMOTION_CONTINUOUS_DYNAMICAL_SYSTEM_FLOATING_BASE_SYSTEM_KINEMATICS_H
+
+#include <memory>
+#include <vector>
+
+#include <BipedalLocomotion/ContinuousDynamicalSystem/DynamicalSystem.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+
+#include <Eigen/Dense>
+
+namespace BipedalLocomotion
+{
+namespace ContinuousDynamicalSystem
+{
+
+// Forward declare for type traits specialization
+class FloatingBaseSystemKinematics;
+}
+}
+
+BLF_DEFINE_CONTINUOUS_DYNAMICAL_SYSTEM_INTERAL_STRUCTURE(
+    FloatingBaseSystemKinematics,
+    (Eigen::Vector3d, Eigen::Matrix3d, Eigen::VectorXd),
+    (Eigen::Vector3d, Eigen::Matrix3d, Eigen::VectorXd),
+    (Eigen::Matrix<double, 6, 1>, Eigen::VectorXd))
```

Comment:
Maybe we can just add a comment naming the types above this macro declaration, so that we don't get confused/spend time searching for types while looking at the documentation below.